sources/api/prairiedog/src/bootconfig.rs

```diff
@@ -117,8 +117,14 @@ fn get_boot_config_settings<P>(config_path: P) -> Result<Option<BootSettings>>
 where
     P: AsRef<Path>,
 {
-    let config_str = fs::read_to_string(config_path.as_ref()).context(error::ReadFileSnafu {
-        path: config_path.as_ref().to_path_buf(),
+    let config_path = config_path.as_ref();
+    if !config_path.try_exists().context(error::ReadFileSnafu {
+        path: config_path.to_path_buf(),
+    })? {
+        return Ok(None);
+    }
+    let config_str = fs::read_to_string(config_path).context(error::ReadFileSnafu {
+        path: config_path.to_path_buf(),
```

Comment:
Commit message could be reworded, it's a bit hard to parse right now.

There's a TOCTOU issue here where `fs::read_to_string` could fail if the file is removed before it runs and after the `try_exists` check. To avoid this, and the redundant check, it might be better not to use `fs::read_to_string` at all.

Reply:
For some context, toml deprecated its from_slice parser, also part of the reason to utilize try_exists above instead of exists() is this very problem, exists is prone to failure on the exact situation you described where try_exists can validate this. Though I wonder it may be better to treat a failure of config_path.try_exists() as equivalent to no file found.

Reply:
Updated per discussion